lib/collection/request-body.js

```diff
@@ -151,12 +173,22 @@ _.assign(RequestBody.prototype, /** @lends RequestBody.prototype */ {
         var mode = this.mode,
             data = mode && this[mode];
 
+        // bail out if there's no data for the selected mode
+        if (!data) {
+            return true;
+        }
+
+        // Handle file mode
         if (mode === RequestBody.MODES.file) {
-            // Handle file mode.
-            return !(data && (data.src || data.content));
+            return !(data.src || data.content);
         }
 
-        return !data || !(_.isString(data) ? data.length : (_.isFunction(data.count) && data.count()));
+        // Handle graphql mode, send empty query anyways
+        if (mode === RequestBody.MODES.graphql) {
```

Comment:
Well... I think we need to check if the body type stores in object format and it has no keys ... like lodash isEmpty of some sort and not hard-code graphql here. This keeps things still manageable when we make future extensions and also keeps in line with present.